src/main/java/com/google/api/codegen/util/SampleConfigSanitizer.java

```diff
@@ -0,0 +1,57 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.util;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.yaml.snakeyaml.Yaml;
+
+public class SampleConfigSanitizer {
+
+  private SampleConfigSanitizer() {
+    // utility class
+  }
+
+  private static final String TYPE_FIELD_NAME = "type";
+  private static final String SCHEMA_VERSION_FIELD_NAME = "config_schema_version";
+  private static final String VALID_TYPE_VALUE =
+      "com.google.api.codegen.samplegen.v1p2.SampleConfigProto";
+  private static final String VALID_SCHEMA_VERSION = "1.2.0";
+
+  // Filter out sample configure files without proper type definition.
+  public static List<String> sanitize(List<String> sampleConfigFileNames) {
```

Comment:
As per [this comment thread](https://docs.google.com/a/google.com/document/d/1g6M5gl2gdrefQGyCW_MHAqZPRBiobCNJ4XChJJ4ufq4/edit?disco=AAAADYbhLxU) the semantics should be that 

* if files are explicitly specified, each such file must be a yaml file that contains at least one sample config doc (otherwise fail)
* if directories are specified, in each such directory we ignore non-yaml files and files that don't include any sample config docs
* across all the sample files and directories specified, we must have obtained at least one sample config doc

Thus, 
* specifying just `--samples=/tmp/non-sample.yaml` will fail
* if we were to support globbing (which is not in the µ-samplegen spec and we should not support it until it is): if `/tmp/foo/`  has both sample-configs and non-sample-config YAML files, specifying `--samples=/tmp/foo/*yaml` will fail, but `--samples=/tmp/foo/` will pass

FYI @software-dov

Reply:
That's done in artman. artman resolves the input and find all the yaml files and pass that to gapic-generator.

Reply:
Also that's for micro-generators.

Reply:
Hmmm... that's OK for now. When artman goes away, we'll need to incorporate that logic into gapic-generator, since that's part of the µgenerator specs. Could you file a FR for this?

Reply:
I don't think the monolith need to support this, because the CLI would be so different anyway.
However, I do agree that it should be supported when monolith works as a protoc Java plugin.

Reply:
Yes, that's what I had in mind. In the protoc world, this is the responsibility of the generators, not  bazel or anything else. Could you file the FR and link it here?

Reply:
Where do you check the third requirement?

Reply:
For now for both cases the generator will emit no samples.

Reply:
issue: https://github.com/googleapis/gapic-generator/issues/2930